File: backend/app/schemas/lancamento_filter_dto.py
from datetime import datetime
from typing import Optional

from app.models.enums import StatusLancamento, TipoLancamento
from app.schemas.query_params import QueryParams


class LancamentoFilterDto(QueryParams):
    data_inicio: Optional[datetime] = None
    data_fim: Optional[datetime] = None
    status: Optional[StatusLancamento] = None
    tipo: Optional[TipoLancamento] = None
    finalidade_id: Optional[int] = None

File: frontend/financeiro-encontro/src/app/services/dto/lancamento-filter.dto.ts
export interface LancamentoFilterDto {
  data_inicio?:   string;
  data_fim?:      string;
  status?:        string;
  tipo?:          string;
  finalidade_id?: number;
  descricao?:     string;
  exclude_ids?:   number[];  // IDs já carregados (lazy load optimization)
}
